tritium-processor/src/main/java/com/palantir/tritium/processor/ProxyAnnotationProcessor.java

```diff
@@ -0,0 +1,426 @@
+        for (MethodElements method : instrumentedMethods) {
+            createMethod(method, specBuilder, methodStaticFields);
+        }
+
+        specBuilder.addMethod(MethodSpec.methodBuilder("of")
+                .addTypeVariables(typeVarNames)
+                .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                .returns(annotatedType)
+                .addParameter(ParameterSpec.builder(InvocationHandler.class, HANDLER_NAME)
+                        .build())
+                .addStatement(
+                        typeVarNames.isEmpty() ? "return new $N($N)" : "return new $N<>($N)", className, HANDLER_NAME)
+                .build());
+
+        if (specBuilder.originatingElements.size() != 1) {
+            messager.printMessage(
+                    Kind.ERROR,
+                    "The generated type must have exactly one originating element: " + specBuilder.originatingElements,
+                    typeElement);
+        }
+        return JavaFile.builder(packageName, specBuilder.build())
+                .skipJavaLangImports(true)
+                .indent("    ")
+                .build();
+    }
+
+    private boolean isMostSpecific(int index, List<MethodElements> methods) {
+        ExecutableType method = methods.get(index).type();
+        for (int j = 0; j < methods.size(); j++) {
+            if (index == j) {
+                continue;
+            }
+            ExecutableType other = methods.get(j).type();
+            if (types.isSubsignature(other, method)) {
+                if (types.isSameType(other.getReturnType(), method.getReturnType())) {
+                    return index < j;
+                }
+                if (types.isSubtype(other.getReturnType(), method.getReturnType())) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @SuppressWarnings("PreferSafeLoggingPreconditions")
+    private static void createMethod(
+            MethodElements method,
+            TypeSpec.Builder typeBuilder,
+            IdentityHashMap<MethodElements, String> methodToStaticFieldName) {
+        String methodName = method.element().getSimpleName().toString();
+        TypeName returnType = TypeName.get(method.type().getReturnType());
+        boolean isVoidMethod = method.type().getReturnType().getKind() == TypeKind.VOID;
+        String parameterString = method.element().getParameters().stream()
+                .map(param -> param.getSimpleName().toString())
+                .collect(Collectors.joining(", "));
+
+        ImmutableSet<String> parameters = method.element().getParameters().stream()
+                .map(param -> param.getSimpleName().toString())
+                .collect(ImmutableSet.toImmutableSet());
+        String throwableName = Parameters.disambiguate("throwable", parameters);
+        String rethrownName = Parameters.disambiguate("throwable", parameters);
```

Comment:
Should this string be `"rethrown"`?

```suggestion
        String rethrownName = Parameters.disambiguate("rethrown", parameters);
```

Reply:
oops, good catch